File: samples/AdventureWorksLT.VB/UdtPhone.vb
Imports DevZest.Data
Imports DevZest.Data.SqlServer
Imports DevZest.Data.Annotations.Primitives
Imports DevZest.Data.Addons
Imports DevZest.Data.SqlServer.Addons

Namespace DevZest.Samples.AdventureWorksLT
    <ModelMemberAttributeSpec(New Type() {GetType(ColumnNotNull), GetType(SqlType)}, True, GetType(_String))>
    Public NotInheritable Class UdtPhone
        Inherits ColumnAttribute

        Protected Overrides Sub Wireup(column As Column)
            If TypeOf column Is _String Then
                column.Nullable(True)
                CType(column, _String).AsSqlNVarChar(25)
            End If
        End Sub
    End Class
End Namespace

File: samples/AdventureWorksLT/UdtPhone.cs
using System;
using DevZest.Data;
using DevZest.Data.SqlServer;
using DevZest.Data.Annotations.Primitives;
using DevZest.Data.Addons;
using DevZest.Data.SqlServer.Addons;

namespace DevZest.Samples.AdventureWorksLT
{
    [ModelMemberAttributeSpec(addonTypes: new Type[] { typeof(ColumnNotNull), typeof(SqlType) }, validOnTypes: new Type[] { typeof(_String) })]
    public sealed class UdtPhone : ColumnAttribute
    {
        protected override void Wireup(Column column)
        {
            if (column is _String stringColumn)
            {
                stringColumn.Nullable(true);
                stringColumn.AsSqlNVarChar(25);
            }
        }
    }
}
